protos/component_metadata/component_metadata.proto

```diff
@@ -0,0 +1,75 @@
+syntax = "proto3";
+
+package mavsdk.rpc.component_metadata;
+
+import "mavsdk_options.proto";
+
+option java_package = "io.mavsdk.component_metadata";
+option java_outer_classname = "ComponentMetadataProto";
+
+// Access component metadata json definitions, such as parameters.
+service ComponentMetadataService {
+
+    /*
+     * Request metadata from a specific component
+     */
+    rpc RequestComponent(RequestComponentRequest) returns(EmptyResponse) { option (mavsdk.options.async_type) = SYNC; }
+
+    /*
+     * Request metadata from the autopilot component
+     */
+    rpc RequestAutopilotComponent(EmptyRequest) returns(EmptyResponse) { option (mavsdk.options.async_type) = SYNC; }
+
+    /*
+     * Register a callback that gets called when metadata is available
+     */
+    rpc SubscribeMetadataAvailable(MetadataAvailableRequest) returns(stream MetadataUpdateResponse) { option (mavsdk.options.async_type) = ASYNC; }
+
+    /*
+     * Request metadata from the autopilot component
+     */
+    rpc GetMetadata(GetMetadataRequest) returns(GetMetadataResponse) { option (mavsdk.options.async_type) = SYNC; }
+}
+
+message RequestComponentRequest {
+    uint32 compid = 1; // The component ID to request
+}
+
+message GetMetadataRequest {
+    uint32 compid = 1; // The component ID to request
+    MetadataType type = 2; // The metadata type
+}
+
+message GetMetadataResponse {
+    MetadataData response = 1;
+}
+
+// Metadata response
+message MetadataData {
+    bool success = 1; // true if the metadata is available
+    string json_metadata = 2; // The JSON metadata
+}
+
+message EmptyRequest {}
+
+message EmptyResponse {}
```

Comment:
Every plugin usually has a Result type that is part of the auto-generation design.

I think you need to return a result for the requests. Best case the result actually tells us if we successfully received the metadata that we requested.

If we don't know which types are populated we could return a response like this:

```
message MetadataResponse {
    Result result = 1;
    bool has_parameter_metadata = 2;
    string parameter_metadata_json = 3;
    ...
}

Reply:
I added the result and clarified what the request is for. Initially I had the requested components in the constructor, but I don't think that's possible through the proto files. Thus it's a different method now.

There's no sync API that blocks and waits for metadata. I'm not sure if that's needed?

Reply:
That's up to you if you want to expose a sync API.